routers/repo/pull.go

```diff
@@ -991,3 +993,40 @@ func CleanUpPullRequest(ctx *context.Context) {
 
 	ctx.Flash.Success(ctx.Tr("repo.branch.deletion_success", fullBranchName))
 }
+
+// DownloadPullDiff render a pull's raw diff
+func DownloadPullDiff(ctx *context.Context) {
+
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Handle(404, "GetIssueByIndex", err)
+		} else {
+			ctx.Handle(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	// Redirect elsewhere if it's not a pull request
+	if !issue.IsPull {
+		ctx.Handle(404, "DownloadPullDiff", fmt.Errorf("Issue is not a pull request"))
+		return
+	}
+
+	pr := issue.PullRequest
+
+	if pr.BaseRepo == nil {
```

Comment:
use `err = pr.GetBaseRepo()` is better.

Reply:
See if you like 8150dbb9 (I'm not sure GetBaseRepo was much better...)